astropy/coordinates/angles.py

```diff
@@ -19,9 +21,46 @@
 
 
 # these are used by the `hms` and `dms` attributes
-hms_tuple = namedtuple("hms_tuple", ("h", "m", "s"))
-dms_tuple = namedtuple("dms_tuple", ("d", "m", "s"))
-signed_dms_tuple = namedtuple("signed_dms_tuple", ("sign", "d", "m", "s"))
+class hms_tuple(NamedTuple):
```

Comment:
Yikes, is this an auto-correction? Why expand a one-liner that is completely obvious?

Reply:
Not an autocorrection. It's `PYI024` (https://beta.ruff.rs/docs/rules/collections-named-tuple/).

The meaning of each element is obvious, but the actual content of each is very not obvious. Did you know that `np.sign()` returns an `int64` (or equivalent), not a plain `int` on scalar input? It's always +/-1 so why not use an int? And are the degrees a float or a Quantity with units? Either seems reasonable. `NamedTuple` exists for precisely this reason: to allow for docstrings and type hints explaining the class.

Reply:
I was mostly just surprised that such large changes are made in a PR that is titled "pre-commit autoupdate"; I think I can fairly easily be convinced the typed version is better, but changes like these should be discussed on their own merit. For instance,  we just removed allowing tuples as an option for input; maybe we should instead deprecate them also for output?

Reply:
Agreed on the sense of moving this to a new PR! Ok. I'm marking this as resolved since we can discuss tuple-or-not in the followup.